TZStackView/TZStackView.swift

```diff
@@ -273,7 +285,23 @@ public class TZStackView: UIView {
         removeArrangedSubview(subview)
     }
 
+    public override func addConstraint(constraint: NSLayoutConstraint) {
+        if !constraintIsAutoGenerated(constraint) {
+            super.addConstraint(constraint)
+        }
+    }
+
+    public override func addConstraints(constraints: [NSLayoutConstraint]) {
+        super.addConstraints(constraints.filter(negate(constraintIsAutoGenerated)))
+    }
+
     override public func updateConstraints() {
+        removeConstraints(constraints.filter(constraintIsAutoGenerated))
+
+        if let superview = superview where superview.constraints.count > 0  {
+            superview.removeConstraints(superview.constraints.filter(constraintIsAutoGenerated))
```

Comment:
Is this a little too aggressive? After all, some of the constraints are for other subviews of the superview. Maybe check if the constraint's first item or second item is `self`?